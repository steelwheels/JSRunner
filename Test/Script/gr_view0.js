/* gr_view0.js */

console.log("* Test Graphics/Primitive Library\n") ;
const primitives = require('Graphics/Primitives') ;
const grcons     = require('Graphics/Console') ;

const pt0  = new primitives.Point(1, 1) ;
const sz0  = new primitives.Size(13, 14) ;
const rct0 = new primitives.Rect(pt0, sz0) ;
const view0 = new primitives.View(rct0) ;

const astr0 = view0.alignStringToLeft(10, "hello") ;
const astr1 = view0.alignStringToRight(10, "hello") ;
const astr2 = view0.alignStringToCenter(10, "hello") ;
console.log("alignLeft:   \"" + astr0 + "\"\n") ;
console.log("alignRight:  \"" + astr1 + "\"\n") ;
console.log("alignCenter: \"" + astr2 + "\"\n") ;

const astr10 = view0.alignStringToLeft(10,   "hello, world !!!") ;
const astr11 = view0.alignStringToRight(10,  "hello, world !!!") ;
const astr12 = view0.alignStringToCenter(10, "hello, world !!!") ;
console.log("alignLeft:   \"" + astr10 + "\"\n") ;
console.log("alignRight:  \"" + astr11 + "\"\n") ;
console.log("alignCenter: \"" + astr12 + "\"\n") ;

/* End of this script */
console.log("Bye") ;

