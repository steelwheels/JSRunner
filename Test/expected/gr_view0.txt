* Test Graphics/Primitive Library
alignLeft:   "hello     "
alignRight:  "     hello"
alignCenter: "  hello   "
alignLeft:   "hello, wor"
alignRight:  "hello, wor"
alignCenter: "hello, wor"
Bye